packages/riverpod_lint/bin/custom_lint.dart

```diff
@@ -813,6 +850,11 @@ Then dispose of the listener when you no longer need the autoDispose provider to
   }
 
   Stream<Lint> _checkValidProviderDeclaration(AstNode providerNode) async* {
+    if (providerNode is ClassDeclaration ||
+        providerNode is FunctionDeclaration) {
+      // Codegen provider
```

Comment:
Unrelated but one thing the linter could do is expose quick fixes, to add things such as:
- automatically adding the "extends _$Foo"
- add the missing `build` method
- when using a function, fix any typo in the Ref type (so no `@riverpod foo(BarRef ref)`)

Reply:
No pressure. :smile: